### PR TITLE
Updating NLU rules to use current thread vs coalesce

### PR DESCRIPTION
**Pull request by morriscode**

### CODE REVIEW

**Review by jkamdjou — COMMENTED** [review]
LGTM - we should prob let test-rules run for a few days and confirm that we see the same results for both current and new rules. do we have a good way to compare?

@alexk307 / @rw-access can y'all confirm that this is the intended usage / we won't see regressions based on the logic implemented?

**Comment by rw-access** [comment]
It's the intended usage, but we might see regressions just because it's new and we're figuring out kinks. But the resulting field will always be a substring of the same coalesce anyway

File: detection-rules/impersonation_employee_urgent_request.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ source: |
 
   and sender.display_name in~ $org_display_names
 
-  and any([body.plain.raw, body.html.inner_text], 
+  and any([body.current_thread.text],
```
Comment (jkamdjou) [inline detection-rules/impersonation_employee_urgent_request.yml:13]: @morriscode this doesn't need an `any` anymore

Comment (morriscode) [reply detection-rules/impersonation_employee_urgent_request.yml]: Roger that, sed didn't catch that one :) thanks!

File: detection-rules/recipients_undisclosed_compauth_check.yml [hunk 2]
```diff
@@ -8,21 +8,24 @@ source: |
...
+      ),
+      (
+          any(body.links, 
+              any([beta.linkanalysis(.)],
+              .credphish.disposition == "phishing"
+              and .credphish.confidence in ("high")
+              )   
           )
       )
     )
```
Comment (aidenmitchell) [inline detection-rules/recipients_undisclosed_compauth_check.yml:31]:
```suggestion
```

**Review by aidenmitchell — CHANGES_REQUESTED** [review]

File: detection-rules/recipients_undisclosed_compauth_check.yml [hunk 3]
```diff
@@ -8,25 +8,26 @@ source: |
...
+          )
+      ),
+      (
+          any(body.links, 
+              any([beta.linkanalysis(.)],
+              .credphish.disposition == "phishing"
+              and .credphish.confidence in ("high")
+              )   
           )
       )
```
Comment (aidenmitchell) [inline detection-rules/recipients_undisclosed_compauth_check.yml:30]:
```suggestion
      )
   )
```

**Comment by yungbeaniebaby** [comment]
is dis gööd to gö?

**Review by aidenmitchell — APPROVED** [review]

**Comment by morriscode** [comment]
> is dis gööd to gö?

I think it is now. One of the earlier PR's introduced another conflict that should be resolved, just need a ✅ to merge

File: detection-rules/body_job_scam_new_sender.yml [hunk 4]
```diff
@@ -4,15 +4,13 @@ description: |
...
-                .name == "financial"
-            )
-          )
+  type.inbound 
+    and any(ml.nlu_classifier(body.current_thread.text).intents,
+      .name in ("job_scam") and .confidence == "high"
+
+      and (
+        any(ml.nlu_classifier(body.current_thread.text).entities, .name == "financial")
+      )
```
Comment (rw-access) [inline detection-rules/body_job_scam_new_sender.yml:13]:
```suggestion
      and any(ml.nlu_classifier(body.current_thread.text).entities, .name == "financial")
```

File: detection-rules/headers_freemail_replyto_returnpath_mismatch.yml [hunk 5]
```diff
@@ -5,7 +5,7 @@ severity: "medium"
 source: |
   type.inbound
   and any(ml.nlu_classifier(body.current_thread.text).intents,
-          .name in ("bec") and .confidence in ("medium", "high")
+    .name in ("bec") and .confidence in ("medium", "high")
```
Comment (rw-access) [inline detection-rules/headers_freemail_replyto_returnpath_mismatch.yml:8]: goodbye, my beautiful formatting

File: detection-rules/headers_replyto_new_domain_nlu_request.yml [hunk 6]
```diff
@@ -15,22 +15,18 @@ source: |
...
   and (
     (
-      any(ml.nlu_classifier(coalesce(body.html.display_text, body.plain.raw)).entities,
-          .name == "financial"
-      )
-      or any(ml.nlu_classifier(coalesce(body.html.display_text, body.plain.raw)).entities,
-             .name == "urgency"
-      )
+        any(ml.nlu_classifier(body.current_thread.text).entities, .name == "financial") 
+        or any(ml.nlu_classifier(body.current_thread.text).entities, .name == "urgency")
```
Comment (rw-access) [inline detection-rules/headers_replyto_new_domain_nlu_request.yml:25]:
```suggestion
        any(ml.nlu_classifier(body.current_thread.text).entities, .name in ("financial", "urgency"))
```

File: detection-rules/impersonation_employee_urgent_request.yml [hunk 7]
```diff
@@ -8,14 +8,19 @@ severity: "medium"
...
+        or (
+          any(ml.nlu_classifier(body.current_thread.text).entities,
+              .name == "urgency" and .name == "request"
+          )
+          and not any(ml.nlu_classifier(body.current_thread.text).intents,
+                      .name == "benign" and .confidence == "high"
           )
+          and not strings.istarts_with(subject.subject, "fwd:")
+        )
+    )
```
Comment (rw-access) [inline detection-rules/impersonation_employee_urgent_request.yml:23]: looks like you had a bug
```suggestion
    any(ml.nlu_classifier(body.current_thread.text).intents,
        .name == "bec" and .confidence == "high"
      )  or (
        any(ml.nlu_classifier(body.current_thread.text).entities,
            .name == "urgency" and .name == "request"
        )
        and not any(ml.nlu_classifier(body.current_thread.text).intents,
                    .name == "benign" and .confidence == "high"
        )
        and not strings.istarts_with(subject.subject, "fwd:")
      )
```

**Comment by morriscode** [comment]
Where thanks @rw-access .. I'm super sick of this pr lol

**Comment by morriscode** [comment]
Maybe I'll just close it all and start over after the formatting update? @rw-access how bad is it?

**Review by rw-access — APPROVED** [review]

**Comment by rw-access** [comment]
All good! Approved!

File: detection-rules/impersonation_employee_urgent_request.yml [hunk 8]
```diff
@@ -8,14 +8,17 @@ severity: "medium"
...
-            and any(ml.nlu_classifier(.).entities, .name == "request")
-            and not any(ml.nlu_classifier(.).intents, .name == "benign" and .confidence == "high")
-            and not strings.istarts_with(subject.subject, "fwd:")
-          )
+  and (
+    any(ml.nlu_classifier(body.current_thread.text).intents, .name == "bec" and .confidence == "high")
+    or (
+      any(ml.nlu_classifier(body.current_thread.text).entities,
+          .name == "urgency" and .name == "request"
+      )
```
Comment (rw-access) [inline detection-rules/impersonation_employee_urgent_request.yml:16]: Shoot I missed this one. These need to be checked separately

Comment (rw-access) [reply detection-rules/impersonation_employee_urgent_request.yml:16]:
```
      any(ml.nlu_classifier(body.current_thread.text).entities, .name == "urgency")
      and any(ml.nlu_classifier(body.current_thread.text).entities, .name == "request")
```

Comment (morriscode) [reply detection-rules/impersonation_employee_urgent_request.yml:16]: I'll submit another or in a little bit to fix sorry!!

**Pull request merged**